plan/types.go

```diff
@@ -37,6 +38,12 @@ type Node interface {
 	// Kind returns the type of procedure represented by this node.
 	Kind() ProcedureKind
 
+	// CallStack returns the list of StackEntry values that created this
+	// Node. A Node may have no associated call stack. This happens
+	// when a Node is constructed from a planner rule and not from a
+	// source location.
+	CallStack() []interpreter.StackEntry
```

Comment:
The planner can add, remove, and mutate plan nodes. Won't that invalidate the stack here?

Reply:
The stack is purely for information. For created nodes, like read filter, it won't exist in this implentation. Replacing the spec will likely keep the same stack if there was one. It's not a problem for a node to have no call stack if it has no textual representation in the source.